src/ORM/Table.php

```diff
@@ -1512,6 +1512,24 @@ public function save(EntityInterface $entity, $options = [])
     }
 
     /**
+     * Try to save an entity and fail if save was not successful
+     *
+     * @param \Cake\Datasource\EntityInterface $entity the entity to be saved
+     * @param array|\ArrayAccess $options The options to use when saving.
+     * @return \Cake\Datasource\EntityInterface
+     * @throws \RuntimeException When the entity couldn't be saved
+     */
+    public function saveOrFail(EntityInterface $entity, $options = [])
+    {
+        $saved = $this->save($entity, $options);
+        if ($saved === false) {
+            throw new RuntimeException(sprintf('Could not save %s because of %s', $entity->source(), print_r($entity->errors(), true)));
```

Comment:
That would be good to be able to catch exceptions specifically related to saving vs the 'les save' if/else way that regular `save()` offers.